deploy-board/deploy_board/webapp/helpers/baseimages_helper.py

```diff
@@ -105,24 +107,32 @@ def get_by_id(request, image_id):
 def get_all_providers(request):
     return rodimus_client.get("/base_images/provider", request.teletraan_user_id.token)
 
-def get_image_events_by_newId_with_result(request, image_id):
-    events = get_image_events_by_newId(request, image_id)
+
+def get_image_update_events_by_new_id(request, image_id):
+    events = rodimus_client.get("/base_images/updates/%s" % image_id, request.teletraan_user_id.token)
 
     for event in events:
-        if event['state'] =='INIT':
-            if event['start_time']:
-                event['result'] = 'UPDATING'
-            else:
-                event['result'] = 'INIT'
-        elif event['state']== 'COMPLETED':
-            if event['error_message']:
-                event['result'] = 'FAILED'
-            else:
-                event['result'] = 'SUCCEEDED'
-        else:
-            event['result'] = event['state']
-    
+        event['status'] = generate_image_update_event_status(event)
+
+    return events
+
+
+def get_image_update_events_by_cluster(request, cluster_name):
+    events = rodimus_client.get("/base_images/updates/cluster/%s" % cluster_name, request.teletraan_user_id.token)
+    for event in events:
+        event['status'] = generate_image_update_event_status(event)
     return events
 
-def get_image_events_by_newId(request, image_id):
-    return rodimus_client.get("/base_images/updates/%s" % image_id, request.teletraan_user_id.token)
+
+def generate_image_update_event_status(event):
+    if event['state'] == 'INIT':
```

Comment:
just curious what are the values of state enum? why do we need to rely on start_time for Updating state?

Reply:
```
/**
 * ┌───────────┐   ┌───────────┐
 * │   INIT    │◄─►│  PAUSED   │
 * └────┬───┬──┘   └─────┬─────┘
 *      │   │            │
 *      │   │      ┌─────▼─────┐
 *      │   └─────►│  ABORTED  │
 *      │          └───────────┘
 * ┌────▼──────┐
 * │ COMPLETED │
 * └───────────┘
 */
public enum BaseImageUpdateState {
    INIT,
    PAUSED,
    COMPLETED,
    ABORTED,
}
```
Details are in https://github.com/pinternal/rodimus/pull/70.